drwmutex.go

```diff
@@ -122,11 +136,15 @@ func (dm *DRWMutex) lockBlocking(isReadLock bool) {
 				copy(dm.writeLocks, locks[:])
 			}
 
-			return
+			return true
+		}
+		if time.Since(start) >= timeout { // Are we past the timeout?
```

Comment:
@balamurugana regarding pushing the timeout down to `newRetryTimerJitter`, it may be (under discussion) that we may want to use different timeout values for different reasons for not acquiring the lock (eg. a file being locked versus write/read quorum not being available).